controllers/PollResponseController.ts

```diff
@@ -0,0 +1,162 @@
+            app.put('/api/users/:uid/poll/:pid',PollResponseController.pollResponseController.updatePollResponse);
+            app.delete('/api/users/:uid/deleteresponse/polls/:pid', PollResponseController.pollResponseController.deletePollResponse);
+        }
+        return PollResponseController.pollResponseController;
+    }
+
+
+    /**
+     * Retrieves all pollResponses from the database for a particular user and returns
+     * an array of pollResponses.
+     * @param {Request} req Represents request from client
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON arrays containing the pollResponse objects
+     */
+    findPollResponsesByUser = (req: Request, res: Response) => {
+        // @ts-ignore
+        let userId = req.params.uid === "my" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
+        // @ts-ignore
+        PollResponseController.pollResponseDao.findPollResponsesByUser(userId)
+            .then((pollResponses: PollResponse[]) => res.json(pollResponses));
+    }
+
+    /**
+     * @param {Request} req Represents request from client, including path
+     * parameter tid identifying the primary key of the pollResponse to be retrieved
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON containing the pollResponse that matches the user ID
+     */
+    findAllUsersReplyPollResponse = (req: Request, res: Response) =>
+        PollResponseController.pollResponseDao.findAllUsersReplyPollResponse(req.params.pid)
+            .then((pollResponses: PollResponse) => res.json(pollResponses));
+
+    /**
+     * @param {Request} req Represents request from client, including path
+     * parameter pid identifying the primary key of the pollResponse to be retrieved
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON containing the pollResponse that matches the poll ID
+     */
+    findPollResponseByPollId = (req: Request, res: Response) =>
+        PollResponseController.pollResponseDao.findPollResponseByPollId(req.params.pid)
+            .then((pollResponses: PollResponse) => res.json(pollResponses));
+
+
+    /**
+     * @param {Request} req Represents request from client, including body
+     * containing the JSON object for the new pollResponse to be inserted in the
+     * database, if the Poll is close, then the response can't be created.
+     * @param {Response} res Represents response to client, including the
+     * body formatted as JSON containing the new pollResponse that was inserted in the
+     * database
+     */
+    createPollResponse = async (req: Request, res: Response) => {
+        // @ts-ignore
+        let userId = req.params.uid === "my" && req.session['profile'] ? req.session['profile']._id : req.params.uid;
+        // let userId = req.session['profile']._id;
+
```

Comment:
We don't need a try catch block her like the one added for updateResponse function? If tested with incorrect pollID and it works then ignore this comment. If not tested please test

Reply:
delete it, but add isPollClosed DAO catch error